diff --git a/scene/dataset_readers.py b/scene/dataset_readers.py
index a5195fa..3822df1 100644
--- a/scene/dataset_readers.py
+++ b/scene/dataset_readers.py
@@ -89,6 +89,15 @@ def readColmapCameras(cam_extrinsics, cam_intrinsics, images_folder, objects_fol
             focal_length_y = intr.params[1]
             FovY = focal2fov(focal_length_y, height)
             FovX = focal2fov(focal_length_x, width)
+        elif intr.model=="SIMPLE_RADIAL" or intr.model=="RADIAL":
+            focal_length_x = intr.params[0]
+            FovY = focal2fov(focal_length_x, height)
+            FovX = focal2fov(focal_length_x, width)
+        elif intr.model=="OPENCV":
+            focal_length_x = intr.params[0]
+            focal_length_y = intr.params[1]
+            FovY = focal2fov(focal_length_y, height)
+            FovX = focal2fov(focal_length_x, width)
         else:
             assert False, "Colmap camera model not handled: only undistorted datasets (PINHOLE or SIMPLE_PINHOLE cameras) supported!"
 
diff --git a/train.py b/train.py
index df70b2d..4137b73 100644
--- a/train.py
+++ b/train.py
@@ -84,10 +84,13 @@ def training(dataset, opt, pipe, testing_iterations, saving_iterations, checkpoi
         image, viewspace_point_tensor, visibility_filter, radii, objects = render_pkg["render"], render_pkg["viewspace_points"], render_pkg["visibility_filter"], render_pkg["radii"], render_pkg["render_object"]
 
         # Object Loss
-        gt_obj = viewpoint_cam.objects.cuda().long()
-        logits = classifier(objects)
-        loss_obj = cls_criterion(logits.unsqueeze(0), gt_obj.unsqueeze(0)).squeeze().mean()
-        loss_obj = loss_obj / torch.log(torch.tensor(num_classes))  # normalize to (0,1)
+        if viewpoint_cam.objects is not None:
+            gt_obj = viewpoint_cam.objects.cuda().long()
+            logits = classifier(objects)
+            loss_obj = cls_criterion(logits.unsqueeze(0), gt_obj.unsqueeze(0)).squeeze().mean()
+            loss_obj = loss_obj / torch.log(torch.tensor(num_classes))  # normalize to (0,1)
+        else:
+            loss_obj = torch.tensor(0.0, device="cuda")
 
         # Loss
         gt_image = viewpoint_cam.original_image.cuda()
diff --git a/utils/camera_utils.py b/utils/camera_utils.py
index 86630a4..c9353f6 100644
--- a/utils/camera_utils.py
+++ b/utils/camera_utils.py
@@ -51,7 +51,7 @@ def loadCam(args, id, cam_info, resolution_scale):
                   FoVx=cam_info.FovX, FoVy=cam_info.FovY, 
                   image=gt_image, gt_alpha_mask=loaded_mask,
                   image_name=cam_info.image_name, uid=id, data_device=args.data_device,
-                  objects=torch.from_numpy(np.array(cam_info.objects)))
+                  objects=torch.from_numpy(np.array(cam_info.objects)) if cam_info.objects is not None else None)
 
 def cameraList_from_camInfos(cam_infos, resolution_scale, args):
     camera_list = []
